src/System.Private.CoreLib/shared/System/Threading/Tasks/TaskContinuation.cs

```diff
@@ -830,15 +822,16 @@ protected static void ThrowAsyncIfNecessary(Exception exc)
             // If unhandled error reporting APIs are available use those, otherwise since this 
             // would have executed on the thread pool otherwise, let it propagate there.
 
-            if (!(exc is ThreadAbortException))
-            {
 #if FEATURE_COMINTEROP
```

Comment:
This should look like this: https://github.com/dotnet/coreclr/blob/614966d1e179c32ce85df2791102847128390ca6/src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncMethodBuilder.cs#L1098